mockwebserver/src/test/java/com/squareup/okhttp/mockwebserver/MockWebServerTest.java

```diff
@@ -255,7 +256,28 @@
     assertTrue(String.format("Request + Response: %sms", elapsedMillis), elapsedMillis <= 1100);
   }
 
-  @Test public void disconnectHalfway() throws IOException {
+  @Test public void disconnectRequestHalfway() throws IOException {
+    server.enqueue(new MockResponse().setSocketPolicy(SocketPolicy.DISCONNECT_DURING_REQUEST_BODY));
+
+    HttpURLConnection connection = (HttpURLConnection) server.getUrl("/").openConnection();
+    connection.setRequestMethod("POST");
+    connection.setDoOutput(true);
+    connection.setFixedLengthStreamingMode(20000);
+    connection.connect();
+    OutputStream out = connection.getOutputStream();
+
+    int i;
+    for (i = 0; i < 20000; i++) {
+      try {
+        out.write('a');
+      } catch (IOException e) {
+        break;
+      }
+    }
+    assertEquals(50f, i / 200f, 2f); // Disconnected at 50% +/- 2%
```

Comment:
Yeah. I certainly can change the test to that. The problem is that on the first go it wrote all 20,000 single-byte writes with no failure.